syntax = "proto3";

service PokemonBattle {
    rpc ChoosePokemon (ChoosePokemonRequest) returns (ChoosePokemonResponse);
    rpc Attack (AttackRequest) returns (AttackResponse);
    rpx GetBattleState (BattleStateRequest) returns (BattleStateResponse);
}

message ChoosePokemonRequest {
    string player_id = 1;
    string pokemon_name = 2;
}

message ChoosePokemonResponse {
    string status = 1;
    string message = 2;
}

message AttackRequest {
    string player_id = 1;
    string move = 2;
}

message AttackResponse {
    string status = 1;
    string move = 2;
}

message BattleStateRequest {
    string game_id = 1;    
}

message BattleStateResponse {
    string player1 = 1;
    string player2 = 2;
    PokemonState pokemon1 = 3;
    PokemonState pokemon2 = 4;
}

message PokemonState {
    string name = 1;
    int32 hp = 2;
    repeated string moves = 3;
}